xnmt/retriever.py

```diff
@@ -30,11 +33,11 @@ def __init__(self, reader, database_file, dev_id_file=None, test_id_file=None):
     self.test_id_file = test_id_file
 
   def __getitem__(self, indices):
-    return Batcher.mark_as_batch(Batcher.pad([self.data[index] for index in indices]))
+    return batcher.mark_as_batch(batcher.pad([self.data[index] for index in indices]))
 
 ##### The actual retriever class
 
-class Retriever(TrainTestInterface):
+class Retriever(GeneratorModel, HierarchicalModel):
```

Comment:
Why do we need to subclass both, although GeneratorModel is already a subclass of HierarchicalModel?